scvi/hub/hub_metadata.py

```diff
@@ -0,0 +1,173 @@
+    HF_LIBRARY_NAME,
+    MODALITY_TAG,
+    MODEL_CLS_NAME_TAG,
+    SCVI_VERSION_TAG,
+    TISSUE_TAG,
+)
+from .model_card_template import template
+
+
+@dataclass_json
+@dataclass
+class HubMetadata:
+    """Placeholder docstring. TODO complete."""
+
+    scvi_version: str
+    anndata_version: str
+    large_data_url: Optional[str] = None
+    model_parent_module: str = DEFAULT_PARENT_MODULE
+
+    @classmethod
+    def from_dir(
+        cls,
+        local_dir: str,
+        anndata_version: str,
+        **kwargs,
+    ):
+        """Placeholder docstring. TODO complete."""
+        torch_model = torch.load(f"{local_dir}/model.pt")
+        scvi_version = torch_model["attr_dict"]["registry_"]["scvi_version"]
+
+        return cls(
+            scvi_version,
+            anndata_version,
+            **kwargs,
+        )
+
+
+@dataclass
+class HubModelCardHelper:
+    """Placeholder docstring. TODO complete."""
+
+    license_info: str
+    model_cls_name: str
+    model_init_params: dict
+    model_setup_anndata_args: dict
+    model_summary_stats: dict
+    model_data_registry: dict
+    scvi_version: str
+    anndata_version: str
+    data_modalities: Optional[List[str]] = None
+    tissues: Optional[List[str]] = None
+    data_is_annotated: Optional[bool] = None
+    data_is_latent: Optional[bool] = None
+    large_data_url: Optional[str] = None
+    model_parent_module: str = DEFAULT_PARENT_MODULE
+    description: str = DEFAULT_MISSING_FIELD
+    references: str = DEFAULT_MISSING_FIELD
+
+    def __post_init__(self):
+        self.data_modalities = self.data_modalities or []
+        self.tissues = self.tissues or []
+        self.model_card = self._to_model_card()
+
+    @classmethod
+    def from_dir(
+        cls,
+        local_dir: str,
+        license_info: str,
+        anndata_version: str,
+        torch_map_location: str = "cpu",
+        data_is_latent: Optional[bool] = None,
+        **kwargs,
+    ):
+        """Placeholder docstring. TODO complete."""
+        torch_model = torch.load(
+            f"{local_dir}/model.pt", map_location=torch_map_location
+        )
+        model_init_params = torch_model["attr_dict"]["init_params_"]
+        registry = torch_model["attr_dict"]["registry_"]
+        model_cls_name = registry["model_name"]
+        scvi_version = registry["scvi_version"]
+        model_setup_anndata_args = registry["setup_args"]
+        model_summary_stats = dict(
+            AnnDataManager._get_summary_stats_from_registry(registry)
+        )
+        model_data_registry = dict(
+            AnnDataManager._get_data_registry_from_registry(registry)
+        )
+
+        # get `is_latent` from the param if it is given, else from adata if it on disk, else set it to None
+        is_latent = data_is_latent
+        if is_latent is None and os.path.isfile(f"{local_dir}/adata.h5ad"):
+            adata = anndata.read_h5ad(f"{local_dir}/adata.h5ad", backed=True)
+            is_latent = _is_latent(adata)
```

Comment:
this can be in some kind of util?

Reply:
since it's only a few lines of code and used only in this spot I think it's fine for now.